mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala

```diff
@@ -94,6 +94,28 @@ private[spark] object SchemaUtils {
   /**
    * Appends a new column to the input schema. This fails if the given output column already exists.
    * @param schema input schema
+   * @param colName new column name. If this column name is an empty string "", this method returns
+   *                the input schema unchanged. This allows users to disable output columns.
+   * @param dataType new column data type
+   * @param nullable Indicates if values of this field can be `null` values.
+   * @param metadata The metadata of this field. The metadata should be preserved during
+   *                 transformation if the content of the column is not modified.
+   * @return new schema with the input column appended
+   */
+  def appendColumn(
```

Comment:
How about adding a default arg to the previous appendColumn method?  It's private, so it does not need to be Java-compatible.